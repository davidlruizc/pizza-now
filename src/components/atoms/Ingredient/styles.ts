import { BACKGROUND_COLOR } from 'assets/styles/colors';
import { Input, InputGroup, Label } from 'reactstrap';
import styled from 'styled-components';

export const IngredientsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
  margin-bottom: 10px;
  padding: 10px 25px;
  border-radius: 10px;
  background-color: ${BACKGROUND_COLOR};
`;

export const IngredientsCounter = styled(InputGroup)`
  width: 150px;
`;

export const IngredientsInput = styled(Input)`
  height: 100%;
  text-align: center;
`;

export const ContentWrapper = styled.div`
  flex-direction: row;
`;

export const Content = styled(Label)`
  margin: 0;
  align-self: center;
  width: 100%;
  font-weight: bold;
`;

export const SubContent = styled(Label)`
  width: 100%;
`;
