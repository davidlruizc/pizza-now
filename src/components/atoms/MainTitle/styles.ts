import { Label } from 'reactstrap';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
`;

export const Title = styled(Label)`
  font-weight: bold;
  font-size: 34px;
  margin-bottom: 0;
`;

export const PizzaIcon = styled.img`
  width: 40px;
  height: 40px;
  margin-left: 15px;
`;
