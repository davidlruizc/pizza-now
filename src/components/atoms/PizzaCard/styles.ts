import { CardImg } from 'reactstrap';
import styled from 'styled-components';

export const CardContainer = styled.div`
  margin-top: 25px;
`;

export const PizzaImage = styled(CardImg)`
  object-fit: cover;
  height: 250px;
`;
