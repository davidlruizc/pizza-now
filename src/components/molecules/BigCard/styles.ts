import { Card, CardBody, Row } from 'reactstrap';
import styled from 'styled-components';

export const BodyCard = styled(CardBody)<{ padding?: boolean; height?: string }>`
  padding: ${(props) => (props.padding ? '1.5rem' : '0')} !important;
  width: 100%;
  height: ${(props) => (props.height ? props.height : '100%')};
`;

export const CardContainer = styled(Card)`
  margin-top: 1em;
  color: black;
`;

export const SubTitle = styled.p`
  margin-bottom: 0;
  opacity: 0.7;
  font-size: 0.8em;
`;

export const CardRow = styled(Row)``;
