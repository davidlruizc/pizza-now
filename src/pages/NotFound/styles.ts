import { Label } from 'reactstrap';
import styled from 'styled-components';
import { device } from 'utils/responsiveTemplate';

export const MainContainer = styled.div`
  height: 80vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const ImageNotFound = styled.img`
  height: 500px;
  object-fit: cover;

  @media ${device.mobile} {
    height: 200px;
  }
`;

export const WrapperContent = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ContentLabel = styled(Label)`
  text-align: center;
`;
