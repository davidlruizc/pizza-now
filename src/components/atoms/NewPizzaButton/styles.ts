import { BACKGROUND_COLOR, PRIMARY_COLOR } from 'assets/styles/colors';
import { Button } from 'reactstrap';
import styled from 'styled-components';

export const ButtonStyled = styled(Button)`
  border-radius: 50%;
  width: 50px;
  height: 50px;
  background-color: ${PRIMARY_COLOR};
  color: ${BACKGROUND_COLOR};
`;
