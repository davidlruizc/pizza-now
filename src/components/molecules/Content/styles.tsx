import { BACKGROUND_CARD } from 'assets/styles/colors';
import styled from 'styled-components';

export const ContentWrapper = styled.div`
  background-color: ${BACKGROUND_CARD};
`;
