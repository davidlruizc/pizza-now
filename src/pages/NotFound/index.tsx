import Content from 'components/molecules/Content';
import Header from 'components/molecules/Header';
import { NavigationUrls } from 'constants/navigationURLS';
import * as React from 'react';
import { useHistory } from 'react-router-dom';
import { ContentLabel, ImageNotFound, MainContainer, WrapperContent } from './styles';
import NoFoundImage from 'assets/images/not-found.svg';
import { Button } from 'reactstrap';

const NotFound: React.FC = () => {
  const history = useHistory();

  const onListPizzas = () => {
    history.push(NavigationUrls.root);
  };

  return (
    <Content>
      <Header onNavigate={onListPizzas} close={true} />
      <MainContainer>
        <ImageNotFound src={NoFoundImage} />
        <WrapperContent>
          <ContentLabel>404 PÃ¡gina no encontrada</ContentLabel>
          <Button onClick={onListPizzas} color="primary">
            Volver al inicio
          </Button>
        </WrapperContent>
      </MainContainer>
    </Content>
  );
};

export default NotFound;
