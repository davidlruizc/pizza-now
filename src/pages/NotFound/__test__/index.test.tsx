import * as React from 'react';
import { create } from 'react-test-renderer';
import { MockProvider } from 'utils/tests/MockProvider';
import NotFound from '../index';
import { ContentLabel } from '../styles';

describe('<NotFound />', () => {
  const component = MockProvider('/random', <NotFound />);

  test('render content', () => {
    expect(component.length).toEqual(1);
  });

  test('verify main content', () => {
    expect(component.find(ContentLabel).text()).toEqual('404 PÃ¡gina no encontrada');
  });
});

describe('NotFound snapshot', () => {
  test('match full UI content', () => {
    const component = create(<NotFound />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
