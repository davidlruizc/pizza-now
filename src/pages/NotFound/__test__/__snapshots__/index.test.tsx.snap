// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound snapshot match full UI content 1`] = `
<div
  className="sc-bdnxRM dHsqWo"
>
  <div
    className="sc-jSFjdj laQnHh container"
  >
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <div
          className="sc-gtsrHT jVtluC"
        >
          <label
            className="sc-dlnjwi breyFc"
          >
            Pizza Now
          </label>
          <img
            alt="pizza-emoji"
            className="sc-hKFxyN WiOO"
            src="pizza.png"
          />
        </div>
      </div>
      <div
        className="col-6"
      >
        <div
          className="sc-gKAaRy iexnav"
        >
          <button
            aria-label={null}
            className="sc-eCApnc iQurul btn btn-secondary btn-lg"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fa fa-times"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-iCoGMd uhlzX"
  >
    <img
      className="sc-fujyAs hKKHrf"
      src="not-found.svg"
    />
    <div
      className="sc-pNWdM gAgbzS"
    >
      <label
        className="sc-jrsJWt hMHKUZ"
      >
        404 PÃ¡gina no encontrada
      </label>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Volver al inicio
      </button>
    </div>
  </div>
</div>
`;
